addons/mail_bot/models/mail_bot.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, _
+import random
+
+
+class MailBot(models.AbstractModel):
+    _name = 'mail.bot'
+
+    def _logic(self, record, values, command=None):
+        odoobot_id = self.env['ir.model.data'].xmlid_to_res_id("mail_bot.partner_odoobot")
+        if len(record) != 1 or values.get("author_id") == odoobot_id:
+            return
+        if self._pinging_odoobot(values) or self._private_channel_with_odoobot(record):
+            body = values.get("body", "").replace(u'\xa0', u' ').strip().lower().strip(".?!")
+            answer = self._answer(record, body, values, command)
+            if answer:
+                message_type = values.get('message_type', 'comment')
+                subtype_id = values.get('subtype_id', self.env['ir.model.data'].xmlid_to_res_id('mail.mt_comment'))
+                record.with_context({"mail_create_nosubscribe": True}).message_post(body=answer, author_id=odoobot_id, message_type=message_type, subtype_id=subtype_id)
+
+    def _answer(self, record, body, values, command=False):
+        # onboarding
+        if self._private_channel_with_odoobot(record):
+            #main flow
+            if body == '😊':
+                return _("Great! :) Did you notice that you can also send attachments, like a picture of your cute dog? Try it!")
+            elif values.get("attachment_ids", False):
+                return _("If you need some assistance, start typing with a <b>\"/\"</b> (i.e. /help) to see some helpful shortcut.")
+            elif command == 'help':
+                return _("Wow you're a natural! As a channel usually contain a lot of users, you can grab the attention with a ping, <b>@Nameoftheuser</b>. Try to ping me with <b>@OdooBot</b>.")
+            elif self._pinging_odoobot(values):
+                return _("That's me! 🎉<br/>") + _("There are 3 different ways in Odoo to interact with your colleagues: <br/>-via this chat window<br/>-via the <a href =\"/web#action=mail.mail_channel_action_client_chat&active_id=%s\">Discuss</a> application:<br/><img src=\"/mail_bot/static/img/odoobot_discuss.png/\"><br/><br/>-or via the chatter:<br/><img src=\"/mail_bot/static/img/odoobot_chatter.png/\"><br/><br/>Enjoy discovering Odoo!") % self.id
+            #easter eggs
+            elif body in ['❤️', _('i love you'), _('love')]:
+                return _("Aaaaaw that's really cute but, you know, bots don't work that way. You're too human for me! Let's keep it professional ❤️")
+            elif body in [_('help'), _('help me'), _('i need help')]:
+                return _("I'm trying to help you, but I'm just a bot... :( You can also check <a href=\"https://www.odoo.com/page/docs\">our documentation</a>) for more information!")
+            elif _('fuck') in body or "fuck" in body:

```

Comment:
because it will be difficult to translate fuck a nice way.  Seems a good idea while talking with AL an ODO. Now I wonder what will happen if some word contains fuck in some language.